mimic/model/identity_objects.py

```diff
@@ -62,17 +80,164 @@ def not_found(message, request):
     return _identity_error_message("itemNotFound", message, NOT_FOUND, request)
 
 
-def forbidden(message, request):
+def unauthorized(message, request):
     """
-    Return a 403 error body associated with a Identity forbidden error.
+    Return a 401 error body associated with a Identity unauthoriazed error.
     Also sets the response code on the request.
 
     :param str message: The message to include in the bad request body.
     :param request: The request on which to set the response code.
 
     :return: dictionary representing the error body.
     """
-    return _identity_error_message("forbidden", message, FORBIDDEN, request)
+    return _identity_error_message("unauthorized", message, UNAUTHORIZED, request)
+
+
+@implementer(IEndpointTemplate, IPlugin)
+class EndpointTemplateStore(object):
+    """
+    A :obj"`EndpointTemplateStore` is an object whiich provides the
+    functionality to integrate an Endpoint Template instance. It is only
+    meant to work with a single Endpoint Template. Use :obj:`ExternalApiStore`
+    to manage multiple Endpoint Templates.
+
+    :ivar list required_mapping: list of entries in the template that are
+        required to be present for a valid template
+    :ivar list optional_mapping: list of entries in the template that are
+        optionally present in a valid template, along with a default value.
+
+
+    Note: The OpenStack documentation[0] does not specify any required
+        parameters. For this implementation, the `id`, `region`, `type`,
+        and `name` fields are required. The `name` and `type` fields
+        are used for creating an instance of the :obj:`ExternalApiStore`
+        that will enable the listing to be put into the Service Catalog;
+        the `id` and `region` fields are used by the :obj:`ExternalApiStore`
+        for managing the templates.
+
+    [0] http://developer.openstack.org/api-ref-identity-v2-ext.html
+    """
+
+    """
+    The following lists of tuples are for mapping the template defined by
```

Comment:
There's nothing wrong with this per se, if it were to be a private attribute (so feel free to ignore this comment) but I vastly prefer a separate, named type that has its own documentation to look at (among other things, sphinx can't "see" this triple-quoted string here, so it's just a comment that will get ignored in generated documentation).

Instead, you can do something like:

```python
@attr.s
class MapInfo(object):
    spec_key = attr.ib()
    attr_name = attr.ib()
    default_value = attr.ib(optional=True)

"..."

required_mapping = [
    MapInfo(*value) for values in [
        ('id', 'id_key'), # et. al.
    ]
]
```